### PR TITLE
Add logging and update process identification "logic"

### PR DESCRIPTION
**Pull request by twimmeh**

### CODE REVIEW

**Comment by twimmeh** [comment]
plz sir. i have worked tirelessly on this software of greatness

File: Software/MainLoop.cs [hunk 1]
```diff
@@ -66,14 +74,16 @@ public static void Run(CancellationToken exitToken)
                 int transferredIn;
                 byte[] readBuffer = new byte[38];
                 ErrorCode ecRead = reader.Transfer(readBuffer, 0, readBuffer.Length, 100, out transferredIn);
-                if (ecRead != ErrorCode.None) throw new Exception("Submit Async Read Failed.");
+                if (ecRead != ErrorCode.None)
+                {
+                    _logger.Error($"Submit Async Read Failed. ErrorCode: {ecRead}");
+                    return;
```
Comment (twimmeh) [inline Software/MainLoop.cs]: this should probably do that cancellation token source thingee

File: Software/MainLoop.cs [hunk 2]
```diff
@@ -118,7 +125,8 @@ public static void Run(CancellationToken exitToken)
                             if (ecWrite != ErrorCode.None)
                             {
                                 // usbReadTransfer.Dispose();
-                                throw new Exception("Submit Async Write Failed.");
+                                _logger.Error($"Submit Async Write Failed on Writer4. ErrorCode: {ecWrite}");
```
Comment (twimmeh) [inline Software/MainLoop.cs]: and here as well

File: Software/MainLoop.cs [hunk 3]
```diff
@@ -137,11 +145,12 @@ public static void Run(CancellationToken exitToken)
                                 if (ecLcdWrite != ErrorCode.None)
                                 {
                                     // usbReadTransfer.Dispose();
-                                    throw new Exception("Submit Async Write Failed.");
+                                    _logger.Error($"Submit Async Write Failed on Writer3. ErrorCode: {ecLcdWrite}");
+                                    return;
```
Comment (twimmeh) [inline Software/MainLoop.cs]: and here

File: Software/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -8,13 +8,13 @@
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
 [assembly: AssemblyTitle("Software")]
-[assembly: AssemblyDescription("")]
+[assembly: AssemblyDescription("Volume mixer of greatness")]
 [assembly: AssemblyConfiguration("")]
 [assembly: AssemblyCompany("")]
 [assembly: AssemblyProduct("Software")]
-[assembly: AssemblyCopyright("Copyright ©  2017")]
+[assembly: AssemblyCopyright("Tober Rowmaker Corp (inc) ©  2017")]
```
Comment (twimmeh) [inline Software/Properties/AssemblyInfo.cs]: IT'S TOBERT! _HONESTLY_

File: Software/MainLoop.cs (hunk 1 above)
Comment (Hueburtsonly) [reply Software/MainLoop.cs]: Good point.

File: Software/MainLoop.cs [hunk 5]
```diff
@@ -153,11 +162,16 @@ public static void Run(CancellationToken exitToken)
                 }
                 else
                 {
-                    Console.WriteLine("Didn't get an interrupt packet?????");
+                    _logger.Warn("Didn't get an interrupt packet?????");
                 }
 
-            } while (!(exitToken.IsCancellationRequested));
+            } while (!(cancellationTokenSource.Token.IsCancellationRequested));
```
Comment (Hueburtsonly) [inline Software/MainLoop.cs]: Add underscore?

File: Software/App.xaml.cs [hunk 6]
```diff
@@ -23,21 +19,24 @@ static void Main(string[] args)
...
+            new Thread(() => MainLoop.Run(cancellationTokenSource, logger)).Start();
+            cancellationTokenSource.Token.Register(() => { Application.Exit(); });
 
             notifyIcon.Visible = true;
 
             // Standard message loop to catch click-events on notify icon
             // Code after this method will be running only after Application.Exit()
             Application.Run();
-
+            logger.Info("Softwware of Greatness exiting.");
```
Comment (Hueburtsonly) [inline Software/App.xaml.cs]: Typo (delete one 'w')

File: Software/MainLoop.cs (hunk 5 above)
Comment (twimmeh) [reply Software/MainLoop.cs]: yarp

**Review by Hueburtsonly — APPROVED** [review]
```
                                     o <------------ Existing code in
                                                     this repository


 
            Good code --> -----------------------







                                     o <------------ This pull request
          .-------------> ------------------------
          |
 The bare minimum code
 standard allowed in this
 repository
```

**Pull request merged**